keylime/common/cli_opts.py

```diff
@@ -0,0 +1,143 @@
+import sys
+
+from keylime import config, keylime_logging, measured_boot
+from keylime.ima import file_signatures, ima
+from keylime.tpm.tpm_abstract import TPM_Utilities
+
+logger = keylime_logging.init_logging("cli_opts")
+
+
+class UserError(Exception):
+    pass
+
+
+def enforce_pcrs(tpm_policy, policy_pcrs, protected_pcrs, pcr_use):
+    policy_pcrs = list(tpm_policy.keys())
+    policy_pcrs.remove("mask")
+
+    for _pcr in policy_pcrs:
+        if int(_pcr) in protected_pcrs:
+            logger.error(
+                'WARNING: PCR %s is specified in "tpm_policy", but will in fact be used by %s. Please remove it from policy',
+                _pcr,
+                pcr_use,
+            )
+            sys.exit(1)
+
+
+def get_opts_error(args):
+    if args.ima_exclude and not args.allowlist:
+        return True, "--exclude cannot be used without an --allowlist"
+    if args.allowlist and args.allowlist_url:
+        return True, "--allowlist and --allowlist-url cannot be specified at the same time"
+    if args.allowlist_url and not (args.allowlist_sig or args.allowlist_sig_url or args.allowlist_checksum):
+        return (
+            True,
+            "--allowlist-url must have either --allowlist-sig, --allowlist-sig-url or --allowlist-checksum to verifier integrity",
+        )
+    if args.allowlist_sig and not (args.allowlist_url or args.allowlist):
+        return True, "--allowlist-sig must have either --allowlist or --allowlist-url"
+    if args.allowlist_sig_url and not (args.allowlist_url or args.allowlist):
+        return True, "--allowlist-sig-url must have either --allowlist or --allowlist-url"
+    if args.allowlist_checksum and not (args.allowlist_url or args.allowlist):
+        return True, "--allowlist-checksum must have either --allowlist or --allowlist-url"
+    if args.allowlist_sig and not args.allowlist_sig_key:
+        return True, "--allowlist-sig must also have --allowlist-sig-key"
+    if args.allowlist_sig_url and not args.allowlist_sig_key:
+        return True, "--allowlist-sig-url must also have --allowlist-sig-key"
+    if args.allowlist_sig_key and not (args.allowlist_sig or args.allowlist_sig_url):
+        return True, "--allowlist-sig-key must have either --allowlist-sig or --allowlist-sig-url"
+    return False, None
+
+
+def process_allowlist(args):
```

Comment:
The rest of this module does seem like cli option handling, but `process_allowlist()` seems very different. I agree with pulling it out of the tenant, but maybe it should be in a separate module designed for policy related things? keylime/policy.pm ?

Reply:
Well, it was sitting within the `tenant` module. But sure, I can certainly create a second module to process inputs to policy (**from the CLI**).